6 feature 6 chat room

1StevenJiang1 opened the pull request.

Dawei-He2002 committed:
Added the CharRoom and Message entities (Feature 6). Code pertaining to the User entity is commented out because that class is not yet implemented.

Dawei-He2002 committed:
Added the interactor for the ChatRoom. However I am still not sure of many things, and a lot of the program is not coded yet, so some lines are commented out.

Dawei-He2002 committed:
Updated the ChatRoomInteractor class to not have extraneous initialization method (i.e. removed initializer, class now uses default object initializer)

Added setter and getter methods to Message

copied two UI classes from the password manager project to the UI package. These are classes I'm sure will be necessary in implementation.

Dawei-He2002 committed:
Deleted Task because that was someone else's entity; amended some comments in ChatRoom.

Dawei-He2002 committed:
Implemented tests for usecase and both entities; MessageTest.testMessageGetAuthor us not yet implemented because it relies on the User entity.

chazzybearr reviewed (changes requested):
src/main/java/entities/ChatRoom.java:
---
> I think the method names AddMessage and GetMessages should be changed to follow java's naming convention (addMessage, getMessage).

src/main/java/entities/Message.java:
---
> In my opinion, when we store the author of a message, we should store the User object, rather than the name of the author. 

> Storing the name is ambiguous, as there could be many users with the same name, and once Message is instantiated, the only information we have on the author of the message is their name, which could lead to problems in the future. I'd be open to discussing this further.

src/test/java/entities/ChatRoomTest.java:
---
> I noticed that the program is creating a new instance of ChatRoom for every test. Although it may seem insignificant, I think putting this in a @ BeforeEach annotation would avoid repeated code.

> The method tearDown doesn't seem to do anything, so it can be removed.

src/test/java/entities/MessageTest.java:
---
> Same points as above.

src/test/java/usecases/ChatRoomInteractorTest.java:
---
> Same points as above.

> I think the test you currently have is a good start, but I think there are too few tests to guarantee that the method works as intended (String edge cases).

---
In general, I think importing wildcards (*) should be removed and replaced with explicit packages instead, as it is considered bad practice. It could cause naming conflicts and is ambiguous.

Dawei-He2002 committed:
Implemented all suggested changes in the code review.

--changed method names to fit with naming conventions
--removed wildcard imports
--added additional tests, removed repeated code and unnecessary code (@After statements) in test files
--Message entity now stores User object instead of user's username; parameter inputs are still unchanged

Dawei-He2002 reviewed (dismissed):
All changes now addressed.

chazzybearr reviewed (dismissed):
Looks good to me!

kldtntg commented on src/main/java/entities/ChatRoom.java:
```diff
@@ -0,0 +1,28 @@
+import java.util.ArrayList;
+
+public class ChatRoom {
+    ArrayList<Message> messages;
```
I think the type here might better be List

nitvishn replied:
Yes, I think that would be more appropriate.

nitvishn reviewed (dismissed):
Other than changing the ArrayList instance variables to List, you're good. I'll approve this so that it can be merged, but you should make this change in the next commit.

Dawei-He2002 committed:
Fixed merge conflicts in Tag.java by git rm command.

Dawei-He2002 committed:
Revert "Implemented all suggested changes in the code review."

This reverts commit fa41f86328d55b528886ec93a0307efe2e8af8eb.

Dawei-He2002 committed:
Merge branch '6-feature-6-chat-room' of https://github.com/CSC207-2022F-UofT/course-project-productive-potato-sloth into 6-feature-6-chat-room
